Fix flaky shopper checkout block creating user with custom password e2e test

veljkho opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #48827
Very flaky instance: https://buildkite.com/organizations/automattic/analytics/suites/woocommerce-core-e2e-tests/tests/bb68dde6-9b7f-88a5-9fbe-6b28f5574c42?branch=all+branches

Looks like it was missing deleting existing user if created during test, now with beforeEach we don't need to think about it for every new test we add.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

```
pnpm test:e2e-pw tests/e2e-pw/tests/shopper/checkout-block.spec.js
```

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

veljkho committed:
Move the customer removal in beforeEach

veljkho committed:
Add changelog

veljkho committed:
No focused tests

adimoldovan commented:
I find it a bit odd to check for customers and deleting them before the test. Why not removing the created test account after the test? Also, if we have 2 tests creating a customer, we should be using different email addresses. Unique ones would be best, like we use all over the e2e suite for generated test data.

veljkho committed:
Update emails for tests and remove before checker

veljkho commented:
@adimoldovan I moved it in `beforeEach` to make freedom of using the same email address for every new test we create. We would use even the same email.

There are many ways to make this test, I ended up creating 2 different emails for tests and removed `beforeEach` checker because with Date the email will be unique, so no need to check if it exists.

adimoldovan commented on plugins/woocommerce/tests/e2e-pw/tests/shopper/checkout-block.spec.js:
```diff
@@ -249,6 +238,17 @@ test.describe(
 							}
 						);
 					}
+					if (
```
Instead of adding a new identical if block, you can update the existing one to

```
if (  [ newAccountEmail, newAccountEmailWithCustomPassword ].includes( response.data[ i ].billing.email ) ) {
...
}
```

veljkho replied:
Nice, done. 👍

veljkho committed:
Improve customer deletion

adimoldovan reviewed (approved):
Thanks @veljkho
